pkg/scenario/stage.go

```diff
@@ -166,6 +185,50 @@ func (c *StageClient) PlanWithSpecificVariableValueToExpect(t *testing.T, option
 	compareValues(t, actualValue, expectedValue, variable)
 }
 
+// PlanAndAssertJSONWithJSONPath performs JSON path planning and assertion in Go testing.
+//
+// t *testing.T: Testing object
+// options *terraform.Options: Terraform options
+// testCases []JSONPathTestCases: Array of JSON path test cases
+func (c *StageClient) PlanAndAssertJSONWithJSONPath(t *testing.T, options *terraform.Options, testCases []JSONPathTestCases) {
+	jsonPlan := terraform.InitAndPlanAndShow(t, options)
+
+	for _, testCase := range testCases {
+		t.Run(testCase.TestName, func(t *testing.T) {
+			var result []interface{}
+			k8s.UnmarshalJSONPath(t, []byte(jsonPlan), testCase.JSONPathToCompare, &result)
```

Comment:
Just curious, is there any specific reason why this library and this particular function from it is required instead of a 'vanilla' Unmarshal of a valid JSON into a go struct/object?

Reply:
I just largerly used it in Terratest since it's a common way to query with JSONPath the plans. The package is still terratest so no big increase in dependencies, and it's maintained/extended upstream.